### ISSUE TITLE
Build fails because aclocal version is greater than 1.16 and the Python build module is missing

### ISSUE THREAD

**Issue by kpeletidis** [nmap/nmap#2913]
While building on OpenSUSE Tumbleweed (aarch64) I see the following error when
I type "make":

```
Compiling libpcre
make[1]: Entering directory '/home/kp/src/git/nmap/libpcre'
CDPATH="${ZSH_VERSION+.}:" && cd . && /bin/sh '/home/kp/src/git/nmap/libpcre/missing' aclocal-1.16 -I m4
/home/kp/src/git/nmap/libpcre/missing: line 81: aclocal-1.16: command not found
WARNING: 'aclocal-1.16' is missing on your system.
         You should only need it if you modified 'acinclude.m4' or
         'configure.ac' or m4 files included by 'configure.ac'.
         The 'aclocal' program is part of the GNU Automake package:
         <https://www.gnu.org/software/automake>
         It also requires GNU Autoconf, GNU m4 and Perl in order to run:
         <https://www.gnu.org/software/autoconf>
         <https://www.gnu.org/software/m4/>
         <https://www.perl.org/>
make[1]: *** [Makefile:1023: aclocal.m4] Error 127
make[1]: Leaving directory '/home/kp/src/git/nmap/libpcre'
make: *** [Makefile:135: build-pcre] Error 2
```

The strange thing is that the "configure" stage didn't indicate any problems. The version of aclocal/automake my system suppports is 1.17 so downgrading is probably not a good idea.

As a workaround I installed the libpcre1 and libpcre-devel packages but even after a "make clean; ./configure" the issue persisted i.e. the configure script kept choosing the libpcre version that comes with nmap. 

I finally worked around the issue with this configuration command:

```
./configure --with-libpcre=/usr/lib64
```

I am not sure why autoconf missed pcre before explicitly telling it where to look. I didn't see this issue with other dependencies. Perhaps something to look into?

Also, the "make" stage failed because the absense of the Python build module wasn't detected. I fixed that by installing the python311-build package. Perhaps this can be checked during configuration?

**Comment by nnposter** [comment]
> I am not sure why autoconf missed pcre before explicitly telling it where to look.

This is a known defect in the current `configure.ac`. See #2907

**Comment by nnposter** [comment]
> Also, the "make" stage failed because the absense of the Python build module wasn't detected.

Presumably you are referring to `setuptools`. If yes, this should be addressed in #2914.

P.S. Please do not commingle separate topics. Capturing them as individual Github issues helps with status tracking.

**Comment by kpeletidis** [comment]
Fixes for #2907 and #2915 verified. Thank you.
